### PR TITLE
Update test tokens

### LINKED ISSUE #79: Update test token naming
Suggest completing with #132

#### Collateral

```
"WSTETH",
"CBETH",
"RETH",
"ARB",
"MAGIC",
```

- [x] Create a list of collateral we will launch with
- [x] ~Update the test tokens to match the same symbols / names as the launch collateral tokens~
- [x] These should still be deployed fresh each time (local and goerli)
- [x] Oracles should be OracleForTestnet

#### OD / ODG

- [x] Update the symbol/name for OD and ODG

### LINKED ISSUE #132: Setup Anvil
- Oracles
     - for local testnet should not rely on any external price feeds
     - For Goerli can still rely on chainlink / camelot pools
- Update package.json to add `yarn deploy:local`
- Add docs in Readme for how to install and run anvil

### PR DESCRIPTION
**Pull request by daopunk**
closes #79 
closes #132 

made sense to do these together^

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -162,71 +184,133 @@ contract DeployGoerli is GoerliParams, Deploy {
...
+    collateralTypes.push(ARB);
+    collateralTypes.push(CBETH);
+    collateralTypes.push(RETH);
+    collateralTypes.push(MAGIC);
   }
 
   function setupPostEnvironment() public virtual override updateParams {
     // deploy Camelot liquidity pool to create market price for OD
-    ICamelotV3Factory(GOERLI_CAMELOT_V3_FACTORY).createPool(address(systemCoin), ARB_GOERLI_WETH);
+    ICamelotV3Factory(GOERLI_CAMELOT_V3_FACTORY).createPool(address(systemCoin), address(collateral[WSTETH]));
```
Comment (pi0neerpat) [inline script/Deploy.s.sol:244]: Please add a comment. Something like:

// This pool will be against WETH or ETH on mainnet. Using WSTETH on testnet for convenience

We def want the ability to swing the OD market price via this pool, so using our own mintable collateral is very helpful here. I'm glad you made this decision.

File: script/Params.s.sol [hunk 2]
```diff
@@ -50,7 +48,7 @@ uint256 constant PROPORTIONAL_GAIN = 222_002_205_862;
 uint256 constant INTEGRAL_GAIN = 16_442;
 
 // Job Params
-uint256 constant JOB_REWARD = 5e18; // 5 HAI
+uint256 constant JOB_REWARD = 5e18; // 5 OD
```
Comment (pi0neerpat) [inline script/Params.s.sol:51]: Please set this to 0.20 OD

File: script/Params.s.sol [hunk 3]
```diff
@@ -26,20 +26,18 @@ import {WAD, RAY, RAD} from '@libraries/Math.sol';
 // --- Utils ---
 
 // to replace OP voting token
-bytes32 constant FTRG = bytes32('FTRG'); // 0x4f50000000000000000000000000000000000000000000000000000000000000
+bytes32 constant ARB = bytes32('ARB'); // 0x4f50000000000000000000000000000000000000000000000000000000000000
```
Comment (pi0neerpat) [inline script/Params.s.sol:29]: - [ ] Please provide the bytes32 string for each collateral for me to update the sdk
- [ ] Please update the comments with the correct values here

File: script/GoerliParams.s.sol [hunk 4]
```diff
@@ -131,12 +131,12 @@ abstract contract GoerliParams is Contracts, Params {
...
-    _safeEngineCParams[WETH].debtCeiling = 100_000_000 * RAD; // 100M COINs
+    _oracleRelayerCParams[WSTETH].safetyCRatio = 1.35e27; // 135%
+    _oracleRelayerCParams[WSTETH].liquidationCRatio = 1.35e27; // 135%
+    _taxCollectorCParams[WSTETH].stabilityFee = RAY + 1.54713e18; // + 5%/yr
+    _safeEngineCParams[WSTETH].debtCeiling = 100_000_000 * RAD; // 100M COINs
 
-    _liquidationEngineCParams[FTRG].liquidationPenalty = 1.2e18; // 20%
-    _collateralAuctionHouseParams[FTRG].maxDiscount = 0.5e18; // -50%
+    _liquidationEngineCParams[ARB].liquidationPenalty = 1.2e18; // 20%
+    _collateralAuctionHouseParams[ARB].maxDiscount = 0.5e18; // -50%
```
Comment (pi0neerpat) [inline script/GoerliParams.s.sol:140]: Why does this list not include all collateral types?

File: script/Registry.s.sol [hunk 5]
```diff
@@ -1,61 +1,70 @@
...
-// --- ARB Goerli ---
-address constant ARB_GOERLI_CHAINLINK_ETH_USD_FEED = 0x62CAe0FA2da220f43a51F86Db2EDb36DcA9A5A08; // ETH to USD
-address constant ARB_GOERLI_CHAINLINK_FTRG_USD_FEED = 0x2eE9BFB2D319B31A573EA15774B755715988E99D; // ARB to USD
-address constant ARB_GOERLI_CHAINLINK_BTC_USD_FEED = 0x6550bc2301936011c1334555e62A87705A81C12C; // ARB to USD
+// --- ARB Mainnet ---
 
-address constant ARB_GOERLI_WETH = 0xEe01c0CD76354C383B8c7B4e65EA88D00B06f36f;
-address constant ARB_GOERLI_GOV_TOKEN = 0x0Ed89D4655b2fE9f99EaDC3116b223527165452D;
+// Deployment params
+address constant DAO_SAFE = address(0); // set this before mainnet deployment
```
Comment (pi0neerpat) [inline script/Registry.s.sol:32]: Thank you

File: script/Registry.s.sol [hunk 6]
```diff
@@ -1,61 +1,70 @@
...
-address constant ARB_WETH = 0x82aF49447D8a07e3bd95BD0d56f35241523fBab1;
-address constant ARB_WSTETH = 0x5979D7b546E38E414F7E9822514be443A4800529;
-address constant ARB_WBTC = 0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f;
-address constant ARB_RETH = 0xB766039cc6DB368759C1E56B79AFfE831d0Cc507;
+// Token contracts (all 18 decimals)
+address constant ARBITRUM_WSTETH = 0x5979D7b546E38E414F7E9822514be443A4800529;
+address constant ARBITRUM_ARB = 0x912CE59144191C1204E64559FE8253a0e49E6548;
+address constant ARBITRUM_CBETH = 0x1DEBd73E752bEaF79865Fd6446b0c970EaE7732f;
+address constant ARBITRUM_RETH = 0xEC70Dcb4A1EFa46b8F2D97C310C9c4790ba5ffA8;
+address constant ARBITRUM_MAGIC = 0x539bdE0d7Dbd336b79148AA742883198BBF60342;
```
Comment (pi0neerpat) [inline script/Registry.s.sol:43]: Sweet thank you for grabbing these

File: script/Registry.s.sol
Comment (pi0neerpat) [inline script/Registry.s.sol:1]: I created an issue to validate all this next chapter https://github.com/open-dollar/od-contracts/issues/153

**Review by pi0neerpat — CHANGES_REQUESTED** [review]
Looking good. Just some small changes

File: script/Params.s.sol (hunk 2 above)
Comment (daopunk) [reply script/Params.s.sol:51]: `
// Job Params
uint256 constant JOB_REWARD = 0.2e18; // 0.2 OD
`

File: script/Deploy.s.sol [hunk 7]
```diff
@@ -112,41 +112,63 @@ contract DeployMainnet is MainnetParams, Deploy {
...
+    collateral[CBETH] = IERC20Metadata(ARBITRUM_CBETH);
+    collateral[RETH] = IERC20Metadata(ARBITRUM_RETH);
+    collateral[MAGIC] = IERC20Metadata(ARBITRUM_MAGIC);
 
-    collateralTypes.push(WETH);
     collateralTypes.push(WSTETH);
+    collateralTypes.push(ARB);
+    collateralTypes.push(CBETH);
+    collateralTypes.push(RETH);
+    collateralTypes.push(MAGIC);
```
[the remark this answers is not in the document]
Comment (daopunk) [reply script/Deploy.s.sol:165]: put ARB on top

File: src/contracts/for-test/MintableVoteERC20.sol
[the remark this answers is not in the document]
Comment (daopunk) [reply src/contracts/for-test/MintableVoteERC20.sol:1]: the linter refers to params in an overriding function. if i change, it changes the function signature, which negates the override

**Review by pi0neerpat — APPROVED** [review]

**Pull request merged**